internal/grpc_dialer.go

```diff
@@ -36,15 +36,12 @@ import (
 )
 
 type (
-	// GRPCDialerParams are passed to GRPCDialer and must be used to create gRPC connection.
-	GRPCDialerParams struct {
+	// ConnectionParameters are passed to GRPCDialer and must be used to create gRPC connection.
+	ConnectionParameters struct {
```

Comment:
Why do we need this `struct` now? It was there because it was part of interface. Now it doesn't. I would just replace it with 3 params.

Reply:
I was going to extend this with an additional client provided object in the next PR. Would it be better to make that a fourth param or keep this object and extend it?

Reply:
It's up to you. Make it `private` at least.